streams/runners/storm/layout/src/main/java/com/hortonworks/streamline/streams/layout/storm/KafkaSourceScheme.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hortonworks.streamline.streams.layout.storm;
+
+import org.apache.storm.spout.MultiScheme;
+
+import java.util.Map;
+
+/**
+ * Scheme to be implemented for Kafka sources to deserialize bytebuffer record into tuples.
+ */
+public interface KafkaSourceScheme extends MultiScheme {
+
+    /**
+     * Key for topic property.
+     */
+    String TOPIC_KEY = "topic";
```

Comment:
We are not trying to enforce all the implementors to use these constants. They can use if they want to. The reason why this configuration is kept as opaque without introducing get/set methods is to avoid adding any methods to interface later like we may want to add reader schema version  also. Other way to do is to have abstract class and have getXXX methods implemented and implementors can use that abstract class if they want to.